### PR TITLE
[6.1] Ac data with missing feature for predict/classify/predict_topk/evaluate.

### PR DESCRIPTION
**Pull request by jakesabathia2**
close https://github.com/apple/turicreate/issues/2844

### CODE REVIEW

File: src/toolkits/activity_classification/ac_data_iterator.cpp [hunk 1]
```diff
@@ -296,6 +296,12 @@ variant_map_type _activity_classifier_prepare_data_aug_impl(
   }
 #endif
 
+  for (auto &feat : features) {
```
Comment (TobyRoseman) [inline src/toolkits/activity_classification/ac_data_iterator.cpp]: It's probably best not to use `auto` here.

Comment (Jarvi-Izana) [reply src/toolkits/activity_classification/ac_data_iterator.cpp]: yeah, flexible_type is enough.

Comment (Jarvi-Izana) [reply src/toolkits/activity_classification/ac_data_iterator.cpp]: sorry, std::string.

Comment (jakesabathia2) [reply src/toolkits/activity_classification/ac_data_iterator.cpp]: very good!

**Pull request merged**